### PR TITLE
Api 811 r4 immunization controller

### PR DESCRIPTION
**Pull request by ilaflamme-va**
Included updated `primarySource` field in the `R4ImmunizationTransformer`.

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  Immunization toFhir() {
+    return Immunization.builder()
+        .resourceType(Immunization.class.getSimpleName())
+        .id(datamart.cdwId())
+        /*
+        There are 2 types of Immunizations. Those that are administered and those reported
+        from an external source "historical". All the ones we currently expose are administered.
+        The primary source for an administered Immunization is true, as the providing this
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:84]: _as the providing this record_
lol... this is my style of grammar.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import gov.va.api.health.r4.api.elements.Extension;
+import gov.va.api.health.uscorer4.api.resources.Immunization;
+import java.util.List;
+import java.util.Optional;
+import lombok.Builder;
+import lombok.NonNull;
+
+@Builder
+final class R4ImmunizationTransformer {
+  @NonNull final DatamartImmunization datamart;
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:21]: Private

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java (hunk 1 above)
Comment (ilaflamme-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:84]: :facepalm:

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java [hunk 3]
```diff
@@ -17,7 +17,11 @@ void empty() {
                 .datamart(DatamartImmunization.builder().build())
                 .build()
                 .toFhir())
-        .isEqualTo(Immunization.builder().resourceType("Immunization").build());
+        .isEqualTo(
+            Immunization.builder()
+                .resourceType("Immunization")
+                .primarySource(Boolean.TRUE)
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java:23]: This don't feel right. 
If there is no other data, shouldnt we be omitting primary source as well?

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java [hunk 4]
```diff
@@ -16,8 +16,9 @@ void empty() {
             R4ImmunizationTransformer.builder()
                 .datamart(DatamartImmunization.builder().build())
                 .build()
-                .toFhir())
-        .isEqualTo(Immunization.builder().resourceType("Immunization").build());
+                .toFhir()
+                .primarySource(null))
+        .isEqualTo(Immunization.builder().resourceType("Immunization").primarySource(null).build());
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java:21]: shouldnt primary source just be removed in this test?

Comment (ilaflamme-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java]: See discussion thread in slack for details.

**Pull request merged**